File: app/src/main/java/com/hades/example/android/qa/mockito/v1/AlertParamGetter2.java
package com.hades.example.android.qa.mockito.v1;

import android.app.Activity;

import com.hades.example.android.R;
import com.hades.example.android.qa.mockito.AlertParam;
import com.hades.example.android.qa.mockito.DefaultUIConfigure;

class AlertParamGetter2 {
    private Activity context;

    public AlertParamGetter2(Activity context) {
        this.context = context;
    }

    public void getAlertParam(AlertParam param, DefaultUIConfigure configure) {
        getDefaultAlertParam(param);
        setCustomAlertParam(param, configure);
    }

    public AlertParam getDefaultAlertParam(AlertParam param) {
        param.setTitle(getDefaultTitle());
        return param;
    }

    public void setCustomAlertParam(AlertParam param, DefaultUIConfigure configure) {
        if (null == configure) {
            return;
        }
        if (null == configure.getDefaultAlertConfigure()) {
            return;
        }
        if (null != configure.getDefaultAlertConfigure().getTitle()) {
            param.setTitle(configure.getDefaultAlertConfigure().getTitle());
        }
    }

    public String getDefaultTitle() {
        return context.getString(R.string.default_title);
    }
}

File: app/src/main/java/com/hades/example/android/qa/v2/AlertParamGetter2.kt
package com.hades.example.android.qa.v2

import android.app.Activity
import com.hades.example.android.R
import com.hades.example.android.qa.AlertParam
import com.hades.example.android.qa.DefaultUIConfigure

internal class AlertParamGetter2(private val context: Activity) {
    fun getAlertParam(param: AlertParam, configure: DefaultUIConfigure?) {
        getDefaultAlertParam(param)
        setCustomAlertParam(param, configure)
    }

    fun getDefaultAlertParam(param: AlertParam): AlertParam {
        param.title = defaultTitle
        return param
    }

    fun setCustomAlertParam(param: AlertParam, configure: DefaultUIConfigure?) {
        if (null == configure) {
            return
        }
        if (null == configure.defaultAlertConfigure) {
            return
        }
        if (null != configure.defaultAlertConfigure!!.title) {
            param.title = configure.defaultAlertConfigure!!.title
        }
    }

    val defaultTitle: String
        get() = context.getString(R.string.default_title)
}
